validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/LocalValidatorSource.java

```diff
@@ -41,29 +41,35 @@
 import tech.pegasys.teku.infrastructure.async.AsyncRunner;
 import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
 import tech.pegasys.teku.spec.Spec;
-import tech.pegasys.teku.validator.api.ValidatorConfig;
+import tech.pegasys.teku.validator.api.KeyStoreFilesLocator;
 
 public class LocalValidatorSource implements ValidatorSource {
 
   private final Spec spec;
-  private final ValidatorConfig config;
+  private final boolean validatorKeystoreLockingEnabled;
   private final KeystoreLocker keystoreLocker;
   private final AsyncRunner asyncRunner;
+  private final KeyStoreFilesLocator keyStoreFilesLocator;
+  private final boolean readOnly;
 
   public LocalValidatorSource(
       final Spec spec,
-      final ValidatorConfig config,
+      final boolean validatorKeystoreLockingEnabled,
       final KeystoreLocker keystoreLocker,
-      final AsyncRunner asyncRunner) {
+      final KeyStoreFilesLocator keyStoreFilesLocator,
+      final AsyncRunner asyncRunner,
+      final boolean readOnly) {
     this.spec = spec;
-    this.config = config;
+    this.validatorKeystoreLockingEnabled = validatorKeystoreLockingEnabled;
     this.keystoreLocker = keystoreLocker;
     this.asyncRunner = asyncRunner;
+    this.keyStoreFilesLocator = keyStoreFilesLocator;
+    this.readOnly = readOnly;
   }
 
   @Override
   public List<ValidatorProvider> getAvailableValidators() {
-    final List<Pair<Path, Path>> filePairs = config.getValidatorKeystorePasswordFilePairs();
+    final List<Pair<Path, Path>> filePairs = keyStoreFilesLocator.parse();
```

Comment:
it might have been ok when it was a private class to return nulls and deal internally, but if it's going to be public, I'm not comfortable with that. I suggest the internals figure out how to pass back an empty list at worst...
most of our code is ensuring nulls aren't passed about, and this feels like it should be similar

Reply:
I guess that would be safe to remove the null validation from the `getAvailableValidators` because the `parse` method will return a list of pair paths, otherwise it will return an empty list.

Reply:
lets not guess, lets prove that it's at least an emptyList. if we guess and we're wrong it's NPE's in random locations...

Reply:
Added a validation in the parse method that ensures never return null.